tests/std/tests/P0067R5_charconv/test.cpp

```diff
@@ -1077,6 +1079,53 @@ void test_right_shift_64_bits_with_rounding() {
     assert(_Right_shift_with_rounding(0xffff'ffff'ffff'ffffULL, 64, false) == 1);
 }
 
+void wchar_tests() {
+    static_assert(sizeof(__DIGIT_TABLE<char>) == sizeof(__DIGIT_TABLE<wchar_t>) / sizeof(wchar_t));
+    auto& fac = use_facet<ctype<wchar_t>>(locale{});
+    for (size_t i = 0; i < sizeof(__DIGIT_TABLE<char>); ++i) {
+        assert(fac.widen(__DIGIT_TABLE<char>[i]) == __DIGIT_TABLE<wchar_t>[i]);
+    }
+
+    wchar_t buffer[32];
+    for (const auto& t : double_to_wide_test_cases) {
+        auto result = __d2s_buffered_n<wchar_t>(begin(buffer), end(buffer), t.value, t.fmt);
+        const wstring_view sv(t.correct);
+        assert(equal(buffer, result.first, sv.begin(), sv.end()));
+    }
+
+    for (const auto& t : float_to_wide_test_cases) {
+        auto result = __f2s_buffered_n<wchar_t>(begin(buffer), end(buffer), t.value, t.fmt);
+        const wstring_view sv(t.correct);
+        assert(equal(buffer, result.first, sv.begin(), sv.end()));
+    }
+
+    wchar_t correct_lo[] = L"0.0001020304";
+    wchar_t correct_hi[] = L"0.0506070809";
+    double val_lo        = 0.0001020304;
+    double val_hi        = 0.0506070809;
+    for (int i = 0; i < 10; ++i) {
+        auto result = __d2fixed_buffered_n(begin(buffer), end(buffer), val_lo, 10);
+        assert(equal(buffer, result.first, correct_lo));
+
+        result = __d2fixed_buffered_n(begin(buffer), end(buffer), val_hi, 10);
+        assert(equal(buffer, result.first, correct_hi));
+
+        val_lo += 0.1010101010;
+        val_hi += 0.1010101010;

```

Comment:
While this is valid (it's succeeding and reliably testing what's intended), it's using a different approach than the rest of the charconv tests in a slightly concerning way. Specifically, it's performing floating-point math with rounded values (`0.1010101010` isn't exactly representable in binary floating-point, so the true value being accumulated is slightly different). The accumulated difference isn't enough to matter, and the "increment the `'0'` characters to `'1'` etc." loop below is independent, but I would recommend using a different approach - that is, simply writing out all of the inputs and outputs in a static table. (This will also simplify the loop, since `lo` and `hi` don't need to be tested in pairs.)